src/cloudHandler.js

```diff
@@ -0,0 +1,46 @@
+var cloudURL;
+var cloudPath;
+
+/**
+ * Default sync cloud handler responsible for 
+ */
+var handler = function (params, success, failure) {
+    if (!cloudPath) {
+        // Default server sync api route
+        cloudPath = '/mbaas/sync/';
+    }
+    var url = cloudURL + cloudPath + params.dataset_id;
+    var json = JSON.stringify(params.req);
+
+    var xhr = new XMLHttpRequest();
+    xhr.open("POST", url, true);
+    xhr.setRequestHeader('Content-type', 'application/json; charset=utf-8');
+    xhr.onreadystatechange = function () {
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            if ((xhr.status >= 200 && xhr.status < 300) || xhr.status === 304) {
+                var responseJson;
+                try {
+                    if (xhr.responseText) {
+                        responseJson = JSON.parse(xhr.responseText);
+                    }
+                } catch (e) {
+                    return failure(e);
+                }
+                success(responseJson);
+            } else {
+                failure(xhr.responseText);
+            }
+        }
+    };
+    xhr.send(json);
```

Comment:
@wei-lee - Sorry for lack of context. This function is not sending this details and that's clear decision - this details are for platform elements that aren't part of sync.
Default handler will be used only for plain sync users - no platform involved.

>fh-sync-js client then need to provide APIs to allow override.

Yes. setCloudHandler method is part now of api to override handler and use it in fh-js-client. For platform integration we will still use fh.cloud and all related things so nothing changing in that manner. Going to create PR for that soon. 

TL;DR - from platform point of view all features will be reflected 1:1.
Community (default) handler will be simple and clean to avoid some legacy stuff.